src/leap/bitmask/services/mail/smtpbootstrapper.py

```diff
@@ -93,11 +92,13 @@ def _download_config_and_cert(self):
         client_cert_path = self._smtp_config.get_client_cert_path(
             self._userid, self._provider_config, about_to_download=True)
 
-        if not is_file(client_cert_path):
+        needs_download = leap_certs.should_redownload(client_cert_path)
+
+        if needs_download:
             # For re-download if something is wrong with the cert
+            # FIXME this doesn't read well. should reword the logic here.
             self._download_if_needed = (
-                self._download_if_needed and
-                not leap_certs.should_redownload(client_cert_path))
+                self._download_if_needed and not needs_download)
```

Comment:
Why do we set `self._download_if_needed` here? I guess this is related to the fixme up there.

This is done after knowing that `needs_download` is True, so `self._download_if_needed` will be always False. That means that the following if is pointless as well as it will never be executed. Can we just remove this assignment and the following `if self._download_if_needed and os.path....`?